src/twisted/application/runner/test/test_runner.py

```diff
@@ -68,11 +68,31 @@ def test_runInOrder(self):
         """
         L{Runner.run} calls the expected methods in order.
         """
-        runner = DummyRunner({})
+
+        calledMethods = []
+
+        self.patch(
```

Comment:
(barging in here too since @glyph seems to have invited trouble from me) As a general guideline, I think I'd say subclassing is better then global patching every time. Every experience I've had with global patching has been fraught when the time comes to refactor in the future, moreso than with subclassing.